eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -957,12 +957,18 @@ protected function getRealAliasId($locationId)
      * @param array $actionMap
      * @param mixed $oldParentAliasId
      * @param mixed $newParentAliasId
+     * @param string[] $alreadyGeneratedAliases
      */
-    protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId)
+    protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId, array &$alreadyGeneratedAliases = [])
```

Comment:
We definitely shouldn't use reference here and it's possible because of reasons mentioned in my last comment.

Reply:
@alongosz solution has been refactored and test has been added.